presto-hive/src/main/java/com/facebook/presto/hive/ConfigBasedCacheQuotaProvider.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.cache.CacheConfig;
+import com.facebook.presto.spi.PrestoException;
+import com.google.inject.Inject;
+import io.airlift.units.DataSize;
+
+import java.util.Optional;
+
+import static com.facebook.presto.hive.CacheQuotaScope.GLOBAL;
+import static com.facebook.presto.hive.CacheQuotaScope.SCHEMA;
+import static com.facebook.presto.hive.CacheQuotaScope.TABLE;
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_UNKNOWN_ERROR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ConfigBasedCacheQuotaProvider
+        implements CacheQuotaProvider
+{
+    private final CacheQuotaScope cacheQuotaScope;
+    private final Optional<DataSize> configuredCacheQuota;
+
+    @Inject
+    public ConfigBasedCacheQuotaProvider(CacheConfig cacheConfig)
+    {
+        requireNonNull(cacheConfig, "cacheConfig is null");
+        this.cacheQuotaScope = cacheConfig.getCacheQuotaScope();
+        this.configuredCacheQuota = cacheConfig.getDefaultCacheQuota();
+    }
+
+    @Override
+    public CacheQuotaRequirement getCacheQuota(String schema, String table)
+    {
+        switch (cacheQuotaScope) {
+            case GLOBAL:
+                return new CacheQuotaRequirement(GLOBAL, configuredCacheQuota);
+            case SCHEMA:
+                return new CacheQuotaRequirement(SCHEMA, configuredCacheQuota);
+            case TABLE:
+                return new CacheQuotaRequirement(TABLE, configuredCacheQuota);
+            default:
+                throw new PrestoException(HIVE_UNKNOWN_ERROR, format("%s is not supported", configuredCacheQuota));
```

Comment:
I think for config based provider, we are still able to support `PARTITION` as a scope?

Reply:
Since we're moving the calculation process from worker to coordinator, and on coordinator, in order to avoid calculation for every split, we then move the logic to `HiveSplitManager.getSplits()`, where we can only access `Schema` and `Table` info. 
So basically we can only support those two for now.

Reply:
For config based, we don't use `schema` and `table` anyway and directly read from config, so we could still keep the same behavior I suppose?